src/spikeinterface/core/baserecordingsnippets.py

```diff
@@ -124,25 +124,32 @@ def set_probes(self, probe_or_probegroup, group_mode='by_probe', in_place=False)
             'Probe must have device_channel_indices'
 
         # this is a vector with complex fileds (dataframe like) that handle all contact attr
-        arr = probegroup.to_numpy(complete=True)
+        probe_as_numpy_array = probegroup.to_numpy(complete=True)
 
         # keep only connected contact ( != -1)
-        keep = arr['device_channel_indices'] >= 0
+        keep = probe_as_numpy_array['device_channel_indices'] >= 0
         if np.any(~keep):
             warn('The given probes have unconnected contacts: they are removed')
 
-        arr = arr[keep]
-        inds = arr['device_channel_indices']
-        order = np.argsort(inds)
-        inds = inds[order]
-
-        # check
-        if np.max(list(inds) + [0]) >= self.get_num_channels():
```

Comment:
@alejoe91 do you happen to remember where does this check come from and what is trying to do?

Reply:
Mmm I'll check tomorrow!

Reply:
This is a protection when a user when to set a probe or probegroups wired (so with device_channel_indices) that do not correcspond to the recording channel count.